src/v/archival/manifest.cc

```diff
@@ -31,40 +34,50 @@
 
 #include <algorithm>
 #include <array>
+#include <chrono>
 
 namespace archival {
 
 manifest::manifest()
   : _ntp()
-  , _rev() {}
+  , _rev()
+  , _last_offset(0) {}
 
 manifest::manifest(model::ntp ntp, model::revision_id rev)
   : _ntp(std::move(ntp))
-  , _rev(rev) {}
+  , _rev(rev)
+  , _last_offset(0) {}
 
-remote_manifest_path manifest::get_manifest_path() const {
+static remote_manifest_path generate_partition_manifest_path(
+  const model::ntp& ntp, model::revision_id rev) {
     // NOTE: the idea here is to split all possible hash values into
     // 16 bins. Every bin should have lowest 28-bits set to 0.
     // As result, for segment names all prefixes are possible, but
     // for manifests, only 0x00000000, 0x10000000, ... 0xf0000000
     // are used. This will allow us to quickly find all manifests
     // that S3 bucket contains.
     constexpr uint32_t bitmask = 0xF0000000;
-    auto path = fmt::format("{}_{}", _ntp.path(), _rev());
-    uint32_t hash = bitmask & murmurhash3_x86_32(path.data(), path.size());
-    return remote_manifest_path(fmt::format(
-      "{:08x}/meta/{}_{}/manifest.json", hash, _ntp.path(), _rev()));
+    auto path = fmt::format("{}_{}", ntp.path(), rev());
+    uint32_t hash = bitmask & xxhash_32(path.data(), path.size());
+    return remote_manifest_path(
```

Comment:
we should add the link saying that we no longer need to randomize the buckets.

Reply:
sure, the gist here is that aws uses prefix for bucket randomization. 

A prefix is the first 2 chars until the next slash. 

`XX.../YY.../` would have the bucket space randomize  as `hash(XX . YY) % bucket_count` internally. 

I was recommended by someone w/ internal knowledge of s3 to break it into multiple prefixes `XX/YY/ZZ/...path` 

@Lazin has done exactly that, but instead did one path prefix randomization. `XXYYZZAA/` - 4 bytes via xxhash32. 

A comment of that form is what I was suggesting.